File: solutions/557.reverse-words-in-a-string-iii/reverse-words-in-a-string-iii.java
class Solution {
    public String reverseWords(String s) {
        String[] arrStr = s.split(" ");
        StringBuilder result = new StringBuilder();
        for(int i = 0; i < arrStr.length; i++) {
            arrStr[i] = new StringBuilder(arrStr[i]).reverse().toString();
            if(i == arrStr.length - 1) result.append(arrStr[i]);
            else result.append(arrStr[i]).append(" ");
        }
        return result.toString();
    }
}

File: solutions/557.reverse-words-in-a-string-iii/reverse-words-in-a-string-iii.cpp
class Solution {
public:
    string reverseWords(string s) {
        auto begin = s.begin();
        for(auto iter = s.begin();iter!=s.end();++iter){
            if(*iter==' '){
                reverse(begin,iter);
                begin = iter+1;
            }
        }
        reverse(begin,s.end());
        
        return s;
    }
};